### PR TITLE
(PA-1874) Add Ubuntu 18.04 platform definition

### PR DESCRIPTION
**Pull request by caseywilliams**
Adds a platform configuration for Ubuntu 18.04, bionic beaver. I also pared down the cases where pl-binutils is required for linux platforms in openssl, since it's not really necessary for newer platforms like this one (we're trying not to have to ship a pl-binutils for 18.04). Will unblock pending green builds on affected platforms.

### CODE REVIEW

**Comment by caseywilliams** [comment]
jenkins please test this with puppet-agent#5.5.x on aix61-POWERfa,aix71-POWERfa,aix72-POWERfa,amazon6-64a,amazon7-64a,arista4-32a,ciscoxr-64a,cumulus25-64a,debian7-32a,debian8-64a,debian9-32a,fedora25-64a,fedora26-64a,fedora27-64a,redhat5-64a,redhat6-32a,redhat7-64a,redhat7-POWERa,redhat7-AARCH64a,redhatfips7-64a,sles11-32a,sles12-64a,sles12-POWERa,solaris10-64a,solaris10-SPARCa,solaris11-64a,solaris11-SPARCa,ubuntu1404-32a,ubuntu1604-64a,ubuntu1604-POWERa

**Comment by caseywilliams** [comment]
jenkins please test this with puppet-agent#5.5.x on aix61-POWERfa,aix71-POWERfa,aix72-POWERfa,amazon6-64a,amazon7-64a,arista4-32a,ciscoxr-64a,cumulus25-64a,debian7-32a,debian8-64a,debian9-32a,fedora25-64a,fedora26-64a,fedora27-64a,redhat5-64a,redhat6-32a,redhat7-64a,redhat7-POWERa,redhat7-AARCH64a,redhatfips7-64a,sles11-32a,sles12-64a,sles12-POWERa,solaris10-64a,solaris10-SPARCa,solaris11-64a,solaris11-SPARCa,ubuntu1404-32a,ubuntu1604-64a,ubuntu1604-POWERa

**Comment by caseywilliams** [comment]
jenkins please test this with puppet-agent#5.5.x on aix61-POWERfa,aix71-POWERfa,aix72-POWERfa,amazon6-64a,amazon7-64a,arista4-32a,ciscoxr-64a,cumulus25-64a,debian7-32a,debian8-64a,debian9-32a,fedora25-64a,fedora26-64a,fedora27-64a,redhat5-64a,redhat6-32a,redhat7-64a,redhat7-POWERa,redhat7-AARCH64a,redhatfips7-64a,sles11-32a,sles12-64a,sles12-POWERa,solaris10-64a,solaris10-SPARCa,solaris11-64a,solaris11-SPARCa,ubuntu1404-32a,ubuntu1604-64a,ubuntu1604-POWERa

File: configs/components/openssl.rb [hunk 1]
```diff
@@ -79,16 +79,15 @@
...
   elsif platform.is_macos?
     pkg.build_requires 'makedepend'
-  elsif platform.is_linux?
-    unless platform.is_fedora? && platform.os_version.delete('f').to_i >= 26
-      # TODO: pdk had this for all linux platforms, but agent didn't - necessary?
-      pkg.build_requires 'pl-binutils'
-    end
+  elsif platform.is_cross_compiled_linux?
+    pkg.build_requires 'pl-binutils'
     pkg.build_requires 'pl-gcc'
```
Comment (ScottGarman) [inline configs/components/openssl.rb]: I'm a bit surprised this was working previously - typically the pl-build-tools package names we've had to specify for cross-compiled builds included the architecture in them, eg, pl-gcc-<arch>. But if that dependency isn't met I'd expect things to blow up quickly.

**Comment by ScottGarman** [comment]
:+1:

**Pull request merged**